src/helpers/spacewalk.ts

```diff
@@ -98,11 +97,11 @@ export function currencyToString(currency: SpacewalkPrimitivesCurrencyId, tenant
     if (stellarAsset.isStellarNative) {
       return 'XLM';
     } else if (stellarAsset.isAlphaNum4) {
-      const code = hex_to_ascii(stellarAsset.asAlphaNum4.code.toHex());
+      const code = hex_to_ascii(stellarAsset.asAlphaNum4.code.toHex()).trim();
       const issuer = convertRawHexKeyToPublicKey(stellarAsset.asAlphaNum4.issuer.toHex());
       return `${code}:${issuer.publicKey()}`;
     } else if (stellarAsset.isAlphaNum12) {
-      const code = hex_to_ascii(stellarAsset.asAlphaNum12.code.toHex());
+      const code = hex_to_ascii(stellarAsset.asAlphaNum12.code.toHex()).trim();
       const issuer = convertRawHexKeyToPublicKey(stellarAsset.asAlphaNum12.issuer.toHex());
       return `${code}:${issuer.publicKey()}`;
     } else {
```

Comment:
Let's maybe wrap the content in the currencyToString functin with a try-catch as well.